src/components/QuickOpenModal.js

```diff
@@ -136,31 +136,30 @@ export class QuickOpenModal extends Component<Props, State> {
     if (item == null) {
       return;
     }
-    const { selectSource, selectedSource, query, searchType } = this.props;
+    const { selectLocation, selectedSource, query, searchType } = this.props;
     if (this.isSymbolSearch()) {
       if (selectedSource == null) {
         return;
       }
-      selectSource(selectedSource.get("id"), {
-        location: {
-          ...(item.location != null ? { line: item.location.start.line } : {})
-        }
+      selectLocation({
+        sourceId: selectedSource.get("id"),
+        ...(item.location != null ? { line: item.location.start.line } : null)
```

Comment:
this is quite hard to read! lets break this into a couple of lines

``` javascript
const line = item.location != null ? item.location.start.line : null;
const location = { sourceId: selectedSource.get("id"), line }
selectLocation(location);
```

also, do we want to have null here instead of an empty object? this may make some things fail

Reply:
Good catch!